test/versioned/google-genai/chat-completions.test.js

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const test = require('node:test')
+const assert = require('node:assert')
+const fs = require('node:fs')
+const path = require('node:path')
+
+const { removeModules } = require('../../lib/cache-buster')
+const { assertSegments, assertSpanKind, match } = require('../../lib/custom-assertions')
+const { assertChatCompletionMessages, assertChatCompletionSummary } = require('./common')
+const GoogleGenAIMockServer = require('./mock-server')
+const helper = require('../../lib/agent_helper')
+
+const {
+  AI: { GEMINI }
+} = require('../../../lib/metrics/names')
+// have to read and not require because @google/genai does not export the package.json
+const { version: pkgVersion } = JSON.parse(
+  fs.readFileSync(path.join(__dirname, '/node_modules/@google/genai/package.json'))
+)
+const { DESTINATIONS } = require('../../../lib/config/attribute-filter')
+const responses = require('./mock-responses')
+const TRACKING_METRIC = `Supportability/Nodejs/ML/Gemini/${pkgVersion}`
+
+test.beforeEach(async (ctx) => {
+  ctx.nr = {}
+  const { host, port, server } = await GoogleGenAIMockServer()
+  ctx.nr.host = host
+  ctx.nr.port = port
+  ctx.nr.server = server
+  ctx.nr.agent = helper.instrumentMockedAgent({
+    ai_monitoring: {
+      enabled: true
+    },
+    streaming: {
+      enabled: true
+    }
+  })
+  const { GoogleGenAI } = require('@google/genai')
+  ctx.nr.client = new GoogleGenAI({
+    apiKey: 'fake-versioned-test-key',
+    vertexai: false,
+    httpOptions: {
+      baseUrl: `http://${host}:${port}/`,
+    },
+    httpMethod: 'GET'
+  })
+})
+
+test.afterEach((ctx) => {
+  helper.unloadAgent(ctx.nr.agent)
+  ctx.nr.server?.close()
+  removeModules('@google/genai')
+})
+
+test('should create span on successful models generateContent', (t, end) => {
+  const { client, agent, host, port } = t.nr
+  helper.runInTransaction(agent, async (tx) => {
+    const result = await client.models.generateContent({
```

Comment:
When executing this in versioned tests, a typeError occurs when `shim`'s `_applyRecorderSegment` function executes `shim.applyContext`.